text/chapter5.md

```diff
@@ -426,66 +381,80 @@ So, constructing values of algebraic data types is simple, but how do we use the
+{-
+TypesDoNotUnify:
+  current = calculateCurrent lightbulb lightbulb
+                             ^^^^^^^^^
+  Could not match type
+    Ohm
+  with type
+    Volt
+-}
+```
 
-1. (Easy) Write a function `circleAtOrigin` which constructs a `Circle` (of type `Shape`) centered at the origin with radius `10.0`.
-1. (Medium) Write a function `doubleScaleAndCenter` which scales the size of a `Shape` by a factor of `2.0` and centers it at the origin.
-1. (Medium) Write a function `shapeText` which extracts the text from a `Shape`. It should return `Maybe String`, and use the `Nothing` constructor if the input is not constructed using `Text`.
+If we instead just used `Number` without `newtype`, then the compiler can't help us catch this mistake:
 
-## Newtypes
+```haskell
+-- This also compiles, but is not as type safe.
+calculateCurrent :: Number -> Number -> Number
+calculateCurrent v r = v / r
+
+battery :: Number
+battery = 1.5
 
-There is an important special case of algebraic data types, called _newtypes_. Newtypes are introduced using the `newtype` keyword instead of the `data` keyword.
+lightbulb :: Number
+lightbulb = 500.0
 
-Newtypes must define _exactly one_ constructor, and that constructor must take _exactly one_ argument. That is, a newtype gives a new name to an existing type. In fact, the values of a newtype have the same runtime representation as the underlying type. They are, however, distinct from the point of view of the type system. This gives an extra layer of type safety.
+current :: Number
+current = calculateCurrent lightbulb lightbulb -- uncaught mistake
+```
 
-As an example, we might want to define newtypes as type-level aliases for `Number`, to ascribe units like pixels and inches:
+This design principle is perhaps better [communicated visually](https://twitter.com/jusrin00/status/875238742621028355/photo/1).
 
+Note that the constructor of a newtype often has the same name as the newtype itself, but this is not a requirement. For example, unique names are also valid:
 ```haskell
-newtype Pixels = Pixels Number
-newtype Inches = Inches Number
+{{#include ../exercises/chapter5/src/ChapterExamples.purs:Watt}}
 ```
 
-This way, it is impossible to pass a value of type `Pixels` to a function which expects `Inches`, but there is no runtime performance overhead.
+In this case, `Watt` is the _type constructor_ and `MakeWatt` is the _data constructor_. These constructors live in different namespaces, even when the names are identical, such as with the `Volt` example. This is true for all ADTs.
 
-Newtypes will become important when we cover _type classes_ in the next chapter, since they allow us to attach different behavior to a type without changing its representation at runtime.
+Another application of newtypes is to attach different _behavior_ to an existing type without changing its representation at runtime. We cover that use case in the next chapter when we discuss _type classes_.
```

Comment:
Does this need more explanation in the next chapter? Or is the note in the Ch6 exercise enough?